run-tests.pl

```diff
@@ -359,6 +359,20 @@ (&)
    }  until => sub { $_[0]->get };
 }
 
+# wrapper around Future::with_cancel which works around an apparent bug:
+#
+# the returned future does not keep a reference to the old future, which means
+# it may get garbage-collected.
+#
+# (also sets the label as a potential debugging aid)
+sub without_cancel {
+   my ($future) = @_;
+   my $new = $future->without_cancel;
+   $new->{oldref} = $future;
+   $new->set_label( "without_cancel(".( $future->label // $future ).")" );
+   return $new;
+}
```

Comment:
Semantic logic here all seems fine (give or take the somewhat gut-wrenchy poking of `{oldref}` but there's not a lot else that could be done here).

Syntax style doesn't match the style guide though. Opening `{` on its own line; whitespace inside `( $future )` and around the `.` operators.

Reply:
Yup sorry. Fixed in 7997669